React/forms.md

```diff
@@ -158,101 +168,95 @@ Now every time the user types into the name input field the `NewStudentForm`'s s
+**Question:** What does the call to `setFormFields` in our event handler do?  Why did we include it?
 
 We can cause our `onFormSubmit` function to be called whenever the form submits by updatting the `render` function by adding an `onSubmit` attribute to the `form` element.
 
 ```javascript
-// NewStudentForm.js
-...
-<form className="new-student-form" onSubmit={this.onFormSubmit}>
-...
+// src/components/NewStudentForm.js
+// ...
+<form className="new-student-form" onSubmit={onFormSubmit}>
+// ...
 ```
 
 ### Callback Function
 
 Now we have a way to detect submit events on the form, but no way to get data to the rest of the application. To solve this we will follow the same strategy we used for event handling previously:
-- Add a function to `StudentCollection` that modifies state (this time by creating a new student)
-- Pass that function as a prop to the `NewStudentForm`
+
+- Add a function to `App` that modifies state (this time by creating a new student)
+- Pass that function as a prop to `StudentCollection` and have `StudentCollection` forward the function to `NewStudentForm`
 - Call the function when the form is submitted
 
 When we're done, our event handling structure will look like this:
 
 ![form submission with callbacks](./images/form-submission-callback.png)
 
-<!-- https://drive.google.com/open?id=1IblXJsKU3EuPJZ9Z9Pjej9UGgxEO4UPL -->
+<!-- https://drive.google.com/file/d/1jagE5V-XT2t81lWNkUviUROTltw6xmjU/view?usp=sharing -->
 
 Does this diagram look familiar? It should!
 
-Our first step is to add a callback function to `StudentCollection` and pass that function to `NewStudentForm` as a prop.
+Our first step is to add a callback function to `App` and pass that function to `NewStudentForm` as a prop.
 
 ```javascript
-// StudentCollection.js
+// src/components/StudentCollection.js
 ...
 // callback function to add students to the list
-addStudent = (student) => {
-  const students = this.state.students;
-  students.push(student);
 
-  this.setState({ students });
-}
+const addStudent = (student) => {
+  // Duplicate the student list.
+  const newStudentList = [...students];
 
-render() {
-  const studentComponents = // ...
-  return (
-    <div>
-      <h3>Students</h3>
-      {studentComponents}
-      <NewStudentForm addStudentCallback={this.addStudent} />
-    </div>
-  );
+  // Find the max id and add 1
+  const nextId = newStudentList.reduce((accumulator, currentStudent) => {
+    return Math.max(accumulator, currentStudent.id);
+  }, 0) + 1;
+
+  newStudentList.push({
+    id: nextId,
+    fullName: student.fullName,
+    email: student.email,
+    present: false,
+  });
+
+  setStudentList(newStudentList);
 }
-...
 ```
 
 Then we can update the `onFormSubmit` function.
 
 ```javascript
 // NewStudentForm.js
-...
-onFormSubmit = (event) => {
-  event.preventDefault();
+// ...
+const onFormSubmit = (event) => {
+    event.preventDefault();
```

Comment:
This function is also over-indented (but I can't suggest easily since it's broken up into additions and deletions).